src/vm/fieldmarshaler.cpp

```diff
@@ -1229,6 +1242,26 @@ BOOL IsStructMarshalable(TypeHandle th)
 }
 
 
+//=======================================================================
+// This function returns TRUE if the field passed in is a fixed buffer.
+// In all other cases it will return FALSE. 
+//=======================================================================
+BOOL IsFixedBuffer(mdFieldDef field, IMDInternalImport* pInternalImport)
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        PRECONDITION(pInternalImport != NULL);
+    }
+    CONTRACTL_END;
+
+    HRESULT hRESULT = pInternalImport->GetCustomAttributeByName(field, g_FixedBufferAttribute, NULL, NULL);
+    return hRESULT == S_OK;
```

Comment:
There is some subtly here. The result of this operation is _not_ an instance of type `BOOL`, but rather the C++ type `bool`. The difference is as follows: `sizeof(BOOL) == 4` and `sizeof(bool) == 1`. In this case there is an implicit conversion, but that is less than ideal most of the time. In this these cases I normally do `hResult == S_OK ? TRUE : FALSE` so types are correct. I am okay with it in this case, but there is a difference and it should be known.